addons/udes_stock/models/stock_picking.py

```diff
@@ -179,6 +186,14 @@ def get_next_picking_name(self, vals, picking_type=None):
                 return get_next_name(self, "stock.picking", sequence=ir_sequence)
         return super().get_next_picking_name(vals, picking_type=picking_type)
 
+    @api.depends('move_lines.product_id', 'move_lines.product_uom_qty', 'move_lines.product_id.weight')
```

Comment:
Don't think that the method gets triggered when product weight changes, it gets called every time on form load and when changing the other 2, but not when changing the weight, because is not a store field and weight is not a field visible in the form view as is inside product form.
```suggestion
    @api.depends('move_lines.product_id', 'move_lines.product_uom_qty')
```